src/main/java/fi/aalto/cs/apluscourses/presentation/exercise/SubmissionResultViewModel.java

```diff
@@ -24,9 +25,23 @@ public String getPresentableName() {
     return getText("presentation.submissionResultViewModel.nameStart") + " " + submissionNumber;
   }
 
+
+  /**
+   * {@summary}Constructs text appearing in parenthesis.
+   *
+   * @return Returns status text for assignment.
+   *         Methods considers case assignment has not yet been graded
+   */
   @NotNull
   public String getStatusText() {
-    return getModel().getPoints() + "/" + getModel().getExercise().getMaxPoints() + " points";
+    SubmissionResult model = getModel();
+    return (model.getStatus() == SubmissionResult.Status.UNKNOWN)
+            ? getText("presentation.submissionResultViewModel.inGrading")
+            : getStatus(model);
+  }
+
+  private String getStatus(SubmissionResult model) {
```

Comment:
Okay, you moved this to a separate method. That's fine. It's now bit weird though that this (non static) method gets the model as a parameter. But it's ok.

Reply:
I wasn't sure what you meant by Inlining the variable so this was my solution and I wanted to avoid calling `getModel()` again, that's why.